textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/utils/Utils.java

```diff
@@ -32,78 +32,86 @@
-                luceneFieldType.setStored(true);
-                luceneFieldType.setStoreTermVectors( true );
-                luceneFieldType.setStoreTermVectorOffsets( true );
-                luceneFieldType.setStoreTermVectorPayloads( true );
-                luceneFieldType.setStoreTermVectorPositions( true );
-                luceneFieldType.setTokenized( true );
-
-                luceneField = new org.apache.lucene.document.Field(
-                        fieldName,(String) fieldValue,luceneFieldType);
-
-                break;
+        switch (fieldType) {
+        case STRING:
+            luceneField = new org.apache.lucene.document.StringField(fieldName, (String) fieldValue, Store.YES);
+            break;
+        case INTEGER:
+            luceneField = new org.apache.lucene.document.IntField(fieldName, (Integer) fieldValue, Store.YES);
+            break;
+        case DOUBLE:
+            double value = (Double) fieldValue;
+            luceneField = new org.apache.lucene.document.DoubleField(fieldName, value, Store.YES);
+            break;
+        case DATE:
+            String dateString = DateTools.dateToString((Date) fieldValue, Resolution.MILLISECOND);
+            luceneField = new org.apache.lucene.document.StringField(fieldName, dateString, Store.YES);
+            break;
+        case TEXT:
+            // By default we enable positional indexing in Lucene so that we can
+            // return
+            // information about character offsets and token offsets
+            org.apache.lucene.document.FieldType luceneFieldType = new org.apache.lucene.document.FieldType();
+            luceneFieldType.setIndexOptions(IndexOptions.DOCS_AND_FREQS_AND_POSITIONS_AND_OFFSETS);
+            luceneFieldType.setStored(true);
+            luceneFieldType.setStoreTermVectors(true);
+            luceneFieldType.setStoreTermVectorOffsets(true);
+            luceneFieldType.setStoreTermVectorPayloads(true);
+            luceneFieldType.setStoreTermVectorPositions(true);
+            luceneFieldType.setTokenized(true);
+
+            luceneField = new org.apache.lucene.document.Field(fieldName, (String) fieldValue, luceneFieldType);
+
+            break;
 
         }
         return luceneField;
     }
+
     /**
-     * @about Creating a new span tuple from span schema, field list 
+     * @about Creating a new span tuple from span schema, field list
      */
-    public static ITuple getSpanTuple( List<IField> fieldList, List<Span> spanList, Schema spanSchema) {
+    public static ITuple getSpanTuple(List<IField> fieldList, List<Span> spanList, Schema spanSchema,
+            boolean removeLastField) {
         IField spanListField = new ListField<Span>(new ArrayList<>(spanList));
         List<IField> fieldListDuplicate = new ArrayList<>(fieldList);
+        // If SourceOperator is of type Keyword, KeywordMatcher also includes a
+        // span which should be discarded by dictionary matcher.

```

Comment:
The purpose of this function `getSpanTuple`, is to add spanList to a tuple **without spanList**. Since this logic is used in many places, so it becomes a separate helper function.
If you don't want span, why do you call this function in the first place? 
If the logic of removing span from a tuple **with spanList** can be used in other places, you can write a new helper function, else you can put this logic in your own code.
The code should be 
```java
removeSpanFromTuple();
getSpanTuple(the new span you need to add);
```